packages/react/src/PageHeader/PageHeader.tsx

```diff
@@ -61,13 +61,18 @@ const hiddenOnNarrow = {
 
 // Root
 // -----------------------------------------------------------------------------
-export type PageHeaderProps = {
-  'aria-label'?: React.AriaAttributes['aria-label']
-  as?: React.ElementType | 'header' | 'div'
-} & SxProp
-
-const Root = React.forwardRef<HTMLDivElement, React.PropsWithChildren<PageHeaderProps>>(
-  ({children, sx = {}, as = 'div'}, forwardedRef) => {
+export type PageHeaderProps = React.PropsWithChildren<
+  {
+    'aria-label'?: React.AriaAttributes['aria-label']
+    as?: React.ElementType | 'header' | 'div'
+    titleVariant?: 'subtitle' | 'medium' | 'large' | ResponsiveValue<'subtitle' | 'medium' | 'large'>
+  } & SxProp
+>
+
+const Root = React.forwardRef(
+  ({children, sx = {}, as = 'div', titleVariant = 'medium', ...rest}: PageHeaderProps, forwardedRef) => {
+    const [fontSize, setFontSize] = useState<string | null | number | string[]>(null)
```

Comment:
This part still causes a layout shift and it is the only way I found to handle [css-in-js style overrides](https://github.com/github/github/blob/master/ui/packages/issue-viewer/components/header/HeaderViewer.tsx#L71)


I wrote my thoughts on the [Slack thread (internal only)](https://github.slack.com/archives/C01L618AEP9/p1717996593042509?thread_ts=1717754823.228669&cid=C01L618AEP9) - I think we should bring some sort of limitation to what we allow as an override especially for PageHeader since the goal of this component is to bring consistency on GH headings.